defuse/src/contract/tokens/nep141/deposit.rs

```diff
@@ -25,34 +26,80 @@ impl FungibleTokenReceiver for Contract {
         amount: U128,
         msg: String,
     ) -> PromiseOrValue<U128> {
-        require!(amount.0 > 0, "zero amount");
+        let amount_value = amount.0;
+        require!(amount_value > 0, "zero amount");
 
-        let msg = if msg.is_empty() {
-            DepositMessage::new(sender_id)
+        let token_id = CoreTokenId::Nep141(Nep141TokenId::new(PREDECESSOR_ACCOUNT_ID.clone()));
+
+        let DepositMessage {
+            receiver_id,
+            action,
+        } = if msg.is_empty() {
+            DepositMessage::new(sender_id.clone())
         } else {
             msg.parse().unwrap_or_panic_display()
         };
 
         self.deposit(
-            msg.receiver_id,
-            [(
-                Nep141TokenId::new(PREDECESSOR_ACCOUNT_ID.clone()).into(),
-                amount.0,
-            )],
+            receiver_id.clone(),
+            [(token_id.clone(), amount_value)],
             Some("deposit"),
         )
         .unwrap_or_panic();
 
-        if !msg.execute_intents.is_empty() {
-            if msg.refund_if_fails {
-                self.execute_intents(msg.execute_intents);
-            } else {
-                // detach promise
-                let _ = ext_intents::ext(CURRENT_ACCOUNT_ID.clone())
-                    .execute_intents(msg.execute_intents);
+        match action {
+            Some(DepositMessageAction::Notify(notify)) => {
+                let mut on_transfer = ext_mt_receiver::ext(receiver_id.clone());
+
+                if let Some(gas) = notify.min_gas {
+                    on_transfer = on_transfer.with_static_gas(gas);
+                }
+
+                let on_transfer = on_transfer.mt_on_transfer(
+                    sender_id,
+                    vec![receiver_id.clone()],
+                    vec![token_id.to_string()],
+                    vec![U128(amount_value)],
+                    notify.msg,
+                );
+
+                let resolution = Self::ext(CURRENT_ACCOUNT_ID.clone())
+                    .with_static_gas(Self::mt_resolve_deposit_gas(1))
+                    .with_unused_gas_weight(0)
+                    .ft_resolve_deposit(&receiver_id, token_id, amount_value);
+
+                on_transfer.then(resolution).into()
```

Comment:
Please, make these variables inline, it's easier to understand chained promises and their returned values this way